File: leet-code/232. Implement Queue using Stacks/ImplementQueueUsingStacks232.java
import java.util.ArrayDeque;
import java.util.Deque;

public class ImplementQueueUsingStacks232 {
    Deque<Integer> stack1 = new ArrayDeque<>();
    Deque<Integer> stack2 = new ArrayDeque<>();

    public void push(int x) {
        stack1.push(x);
    }

    public int pop() {
        while (!stack1.isEmpty()) {
            stack2.push(stack1.pop());
        }
        int answer = stack2.pop();
        while (!stack2.isEmpty()) {
            stack1.push(stack2.pop());
        }
        return answer;
    }

    public int peek() {
        while (!stack1.isEmpty()) {
            stack2.push(stack1.pop());
        }
        int answer = stack2.element();
        while (!stack2.isEmpty()) {
            stack1.push(stack2.pop());
        }
        return answer;
    }

    public boolean empty() {
        return stack1.isEmpty();
    }
}

File: leet-code/232. Implement Queue using Stacks/ImplementQueueUsingStacks232.kt
package com.example

import java.util.LinkedList

class ImplementQueueUsingStacks232 {
    val stack = LinkedList<Int>()
    val stackHelper = LinkedList<Int>()

    fun push(x: Int) {
        stack.push(x)
    }

    fun pop(): Int {
        while (!stack.isEmpty()) {
            stackHelper.push(stack.pop())
        }
        val item = stackHelper.pop()
        while (!stackHelper.isEmpty()) {
            stack.push(stackHelper.pop())
        }
        return item
    }

    fun peek(): Int {
        while (!stack.isEmpty()) {
            stackHelper.push(stack.pop())
        }
        val item = stackHelper.peek()
        while (!stackHelper.isEmpty()) {
            stack.push(stackHelper.pop())
        }
        return item
    }

    fun empty(): Boolean {
        return stack.isEmpty()
    }
}
